atomic_reactor/util.py

```diff
@@ -852,4 +853,226 @@ def df_parser(df_path, workflow=None, cache_content=False, env_replace=True, par
+
+    def get_old_labels_source(self):
+        return self.get_old_labels(self.LABEL_TYPE_SOURCE)
+
+    def get_old_labels_component(self):
+        return self.get_old_labels(self.LABEL_TYPE_COMPONENT)
+
+    def get_old_labels_host(self):
+        return self.get_old_labels(self.LABEL_TYPE_HOST)
+
+    def get_all_labels(self, label_type):
+        """returns tuple with all label names"""
+        return self._label_names[label_type]
+
+    def get_all_labels_name(self):
+        return self.get_all_labels(self.LABEL_TYPE_NAME)
+
+    def get_all_labels_version(self):
+        return self.get_all_labels(self.LABEL_TYPE_VERSION)
+
+    def get_all_labels_release(self):
+        return self.get_all_labels(self.LABEL_TYPE_RELEASE)
+
+    def get_all_labels_arch(self):
+        return self.get_all_labels(self.LABEL_TYPE_ARCH)
+
+    def get_all_labels_vendor(self):
+        return self.get_all_labels(self.LABEL_TYPE_VENDOR)
+
+    def get_all_labels_source(self):
+        return self.get_all_labels(self.LABEL_TYPE_SOURCE)
+
+    def get_all_labels_component(self):
+        return self.get_all_labels(self.LABEL_TYPE_COMPONENT)
+
+    def get_all_labels_host(self):
+        return self.get_all_labels(self.LABEL_TYPE_HOST)
+
+    def get_preferred_label(self, label_type):
+        """
+        returns the most preferred label name
+        if there isn't any correct name in the list
+        it will return newest label name
+        """
+        for lbl in self._label_names[label_type]:
+            if lbl in self._df_labels:
+                return lbl
+        return self._label_names[label_type][0]
+
+    def get_pref_label_name(self):
+        return self.get_preferred_label(self.LABEL_TYPE_NAME)
+
+    def get_pref_label_version(self):
+        return self.get_preferred_label(self.LABEL_TYPE_VERSION)
+
+    def get_pref_label_release(self):
+        return self.get_preferred_label(self.LABEL_TYPE_RELEASE)
+
+    def get_pref_label_arch(self):
+        return self.get_preferred_label(self.LABEL_TYPE_ARCH)
+
+    def get_pref_label_vendor(self):
+        return self.get_preferred_label(self.LABEL_TYPE_VENDOR)
+
+    def get_pref_label_source(self):
+        return self.get_preferred_label(self.LABEL_TYPE_SOURCE)
+
+    def get_pref_label_component(self):
+        return self.get_preferred_label(self.LABEL_TYPE_COMPONENT)
+
+    def get_pref_label_host(self):
+        return self.get_preferred_label(self.LABEL_TYPE_HOST)
+
+    def get_new_names_by_old(self):
+        """Return dictionary, new label name indexed by old label name."""
+        newdict = {}
+
+        for label_type in self._label_names:
+            for oldname in self._label_names[label_type][1:]:
+                newdict[oldname] = self._label_names[label_type][0]
+        return newdict
+
+    def get_value_only(self, label_type):
+        """
+        returns label value
+        Raises KeyError if label doesn't exist
+        """
+        return self._label_values[label_type]
```

Comment:
What's this for? We don't need this if we have a method that returns both the label name and its value.